src/util.c

```diff
@@ -623,3 +623,41 @@ bool p11prov_x509_names_are_equal(CK_ATTRIBUTE *a, CK_ATTRIBUTE *b)
     X509_NAME_free(xb);
     return cmp == 0;
 }
+
+char *p11prov_alloc_sprintf(int size_hint, const char *format, ...)
+{
+    char *buf = NULL;
+    va_list args;
+    int repeat = 1;
+    int ret;
+
+    va_start(args, format);
```

Comment:
Might need a `va_copy` so the following loop work?

By the way, here is a BSD licensed implementation of vasprintf:
https://github.com/p11-glue/p11-kit/blob/d39043f7c6e44247b5b1a237888e80b2a4d9c2b2/common/compat.c#L618

Reply:
another nice catch, I wish OpenSSL had a BIO function instead of trying to implement my own ...

Reply:
In this case all arguments would be consumed though,
so I wonder if just putting the va_start()/va_end() around the vsnpritf call isn't a simpler option